### ISSUE TITLE
Running jlink task with any version of Gradle >6.3 results in error

### ISSUE THREAD

**Issue by mjparme** [beryx/badass-jlink-plugin#177]
I can run the jlink task for Gradle version 6.3 and prior. However, once I got to 6.4 or above the jlink task throws an error. I have tried all versions of Gradle from 6.4 up to the current 6.8.2. I have also tried with several JDK versions (11, 12,13,14,15) and the JDK version has no effect. As long as it is Gradle 6.3 or prior all is well regardless of JDK version. Likewise if it is Gradle 6.4 or above the error occurs regardless of JDK version. I am using version 2.23.5 of the plugin (I had also previously tested 2.23.3 and had the same experience)

The error:

```
> Task :jlink FAILED
The module name specified in 'application.mainModule' (null) has not the expected value (com.example.skeleton).
Error: Error reading module: /Users/MP250590/projects/java/badass-jlink-bug/build/jlinkbase/jlinkjars/badass-jlink-bug.jar


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jlink'.
> Process 'command '/Users/MP250590/.sdkman/candidates/java/15.0.2-open/bin/jlink'' finished with non-zero exit value 1
```
I have attached two test projects to demonstrate the problem. If you untar the one with gradle 6.3 in its name and execute `./gradlew jlink` all will be well.

If you untar the one with gradle 6.4 in its name and execute `./gradlew jlink` the error will occur. I have attached the stacktrace for this and also provided the output of `./gradlew --info`

The documentation seems to suggest that the plugin works with "Gradle 4.8 or newer". Is this inaccurate?

[badass-jlink-bug-gradle-6.3.tar.gz](https://github.com/beryx/badass-jlink-plugin/files/6213892/badass-jlink-bug-gradle-6.3.tar.gz)
[badass-jlink-bug-gradle-6.4.tar.gz](https://github.com/beryx/badass-jlink-plugin/files/6213893/badass-jlink-bug-gradle-6.4.tar.gz)
[stacktrace.txt](https://github.com/beryx/badass-jlink-plugin/files/6213914/stacktrace.txt)
[info.txt](https://github.com/beryx/badass-jlink-plugin/files/6213930/info.txt)

**Comment by siordache** [comment]
Solution 1:
Replace
```
  id 'org.openjfx.javafxplugin' version '0.0.8'
```
with
```
  id 'org.openjfx.javafxplugin' version '0.0.9'
```

Solution 2:
Replace
```
mainClassName = "$moduleName/com.example.skeleton.ClassWithMain"
```
with
```
mainClassName = "com.example.skeleton.ClassWithMain"
```

**Comment by mjparme** [comment]
Thanks for this. When I first ran into this issue a couple of months back I had checked for a new version of the javafxplugin and `0.0.8` was the newest at that time. I decided to revisit this a few days ago and decided to actually open an issue rather than procrastinate. And this time I did indeed forget to check to see if there was a newer version of the javafxplugin before opening the issue.

I actually had to do both proposed solutions. Upgrading to `0.0.9` fixed the problem where the task would fail but the resulting image would not run. The mainClassName was `com.example.skeleton/com.example.skeleton/com.example.skeleton.ClassWithMain` so I had to also remove the `$moduleName` variable from the mainClassName which then resulted in the correct mainClassName in the startup script in the image i.e. `com.example.skeleton/com.example.skeleton.ClassWithMain`.

Interestingly the JavaFX getting starting example for a modular app shows the $moduleName variable so that doc must just be outdated (https://github.com/openjfx/samples/blob/master/CommandLine/Modular/Gradle/hellofx/build.gradle)

Thanks for this info, everything is working fine now with Java 15 and Gradle 6.8.2.

For the benefit of any searchers that stumble on this the full working build.gradle is:

```groovy
plugins {
  id 'application'
  id 'org.openjfx.javafxplugin' version '0.0.9'
  id 'org.beryx.jlink' version '2.23.5'
}

repositories {
    mavenCentral()
}

javafx {
    version = "13"
    modules = [ 'javafx.controls' ]
}

mainClassName = "com.example.skeleton.ClassWithMain"

jlink {
    options = ['--strip-debug', '--compress', '2', '--no-header-files', '--no-man-pages']
    launcher {
        name = 'myProject'
    }
}
```

The resulting app can be executed with: `./build/image/bin/myProject`
